pkg/core/data/transactions/provider.go

```diff
@@ -226,6 +229,38 @@ func (p *provider) GetBalance(ctx context.Context, vk ViewKey) (uint64, error) {
 	return res.Balance, nil
 }
 
+func (p *provider) CalculateMempoolBalance(ctx context.Context, vkBytes []byte, txs []ContractCall) (uint64, error) {
```

Comment:
I feel that something so Mempool specific does not belong in the `Provider` interface. I'd rather change the name of `Verifier` interface to `UnconfirmedTxProber` and add `CalculateBalance` there (without `Mempool` which is the component using the prober). This has the advantage that we do not need to inject two different interfaces and testing/mocking would be less verbose

Reply:
I implemented this one myself :)